### DISCUSSION TITLE
HOWTO] long press

### DISCUSSION THREAD

**Discussion by hoshi411** [waydroid/waydroid#929] [General]
waydroid on manjaro gnome laptop.

I have tried several different mouse button combos but cannot open the standard android long press contextual menus.  Usually when you long press on apps, a contextual menu is expected to appear but long click with the mouse does nothing .  Middle click and right click also do nothing. Key combos do nothing, such as alt+click or ctrl+click etc.

**Comment by donjoe0** [comment]
Tried it on a tablet with unrecognized touchscreen but having a working touchpad on the attached keyboard: keeping the equivalent of "left click" long-pressed sends a long-touch to the Android in Waydroid, so in the touchpad case I do a quick double-tap (equivalent to a left-click if you're about to start a drag or swipe action) but after the second tap I keep the finger on the touchpad for a longer time before starting any other movements. This brings up the expected context menu, and then I can lift the finger and choose something to tap in the menu, or just drag the finger from where I left it e.g. if this was a homescreen shortcut that I wanted to reposition or drag away to the Remove button etc.

I don't know if this a dependency that made the action possible, but I also have **waydroid-helper** installed, and I've enabled `multi_windows` and `cursor_on_subsurface` in the Settings / Props.

This is all on Kubuntu 22.04 in Waydroid 1.6.1 with LineageOS 18 (GApps), installed manually to replace the v20 that got installed by default.
